cpp_exercises/day2/date/date.h

```diff
@@ -28,13 +28,30 @@ class Date
     int year() const {return m_year;}
 
 private:
+
+    enum MonthZeroBase {
+        JanuaryZeroBase = 0,
+        FebruaryZeroBase,
+        MarchZeroBase,
+        AprilZeroBase,
+        MayZeroBase,
+        JuneZeroBase,
+        JulyZeroBase,
+        AugustZeroBase,
+        SeptemberZeroBase,
+        OctoberZeroBase,
+        NovemberZeroBase,
+        DecemberZeroBase
+    };
+
     static bool isLeapYear(int year);
+    static int getMonthDays(int month, int year);
 
     int m_day;
     int m_month;
     int m_year;
 
-    int m_monthsDays[12] = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
+    constexpr static int m_monthsDays[12] = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
```

Comment:
Piccola nota sul naming: le costanti io sono solito nominarle con nomi interamente maiuscoli, e in questo caso trattandosi di una non-variabile-membro non userei il prefisso `m_`.
In sostanza la chiamerei `MONTHS_DAYS`, seguendo un po' quello che è lo stile di una `#define`, ma con la sicurezza aggiunta dal type-system del linguaggio.
Non so se Luca abbia un pattern di naming diverso per le costanti a tempo di compilazione.